### PR TITLE
Fix/5

### PR DESCRIPTION
**Pull request by sdtsui**
- Fixes non-deterministic bugs on OSX.
- Needs linting/cleanup. Can create a stack with #4 and close in tandem.

### CODE REVIEW

**Comment by sdtsui** [comment]
re: coverage report /\

This shows that we actually hit L88 of index.js locally, even though coverage [shows](https://coveralls.io/builds/14266015/source?filename=index.js#L88) we don't. Interesting. Will look into this next.
```
> libp2p-gossip-discovery@0.0.0 test /Users/sdt/dfinity/js-libp2p-gossip-discovery
> node ./tests/index.js

TAP version 13
# tests
  discovery:gossip connected peer, restarting discovery +0ms
peerBookToJson
peerBookToJson
readPeers
filterPeers
  discovery:gossip connected peer, restarting discovery +10ms
peerBookToJson
readPeers
filterPeers
ok 1 (unnamed assert)
node id match - returning early in _peerDiscovery
I have myself.
  discovery:gossip connected peer, restarting discovery +7ms
peerBookToJson
peerBookToJson
readPeers
filterPeers
ok 2 (unnamed assert)
  discovery:gossip connected peer, restarting discovery +6ms
peerBookToJson
readPeers
filterPeers
ok 3 (unnamed assert)
  discovery:gossip already have peer  QmUct65PwhFdGoPvYuKmDaQedDG7pveGgznee3P2sACSWB +7ms
# erros - no data sent
  discovery:gossip connected peer, restarting discovery +5s
  discovery:gossip connected peer, restarting discovery +11ms
readPeers
ok 4 should be equal
# erros - invalid len
  discovery:gossip connected peer, restarting discovery +5s
readPeers
ok 5 should be equal

1..5
# tests 5
# pass  5

# ok
```

File: index.js [hunk 1]
```diff
@@ -60,6 +60,15 @@ module.exports = class handlePeers extends EventEmitter {
...
   _onConnection (peer) {
+    if (this.node.peerInfo.id._idB58String === peer.id._idB58String) {
+      console.log("node id match - returning early in _onConnection")
+      if (node.peerBook.has(peer)) {
+        console.log("I have myself.")
+      }
+      this.node.peerBook.remove(peer)
+      return
+    }
+
```
Comment (wanderer) [inline index.js]: Is this second check needed? We shouldn't connect to our self if you filter below.

File: index.js [hunk 2]
```diff
@@ -75,6 +84,15 @@ module.exports = class handlePeers extends EventEmitter {
...
       newPeers.forEach(peer => {
+        if (node.peerInfo.id._idB58String === peer.id._idB58String) {
+          console.log("node id match - returning early in _peerDiscovery")
+          if (node.peerBook.has(peer)) {
+            console.log("I have myself.")
+          }
+          node.peerBook.remove(peer)
+          return
+        }
+
```
Comment (wanderer) [inline index.js]: So if this works right, then we should never connect to ourself?

**Comment by sdtsui** [comment]
Fixed in the following commits:
https://github.com/dfinity/js-libp2p-gossip-discovery/commit/3b101b71a6f06d610825d5153d9f0159d292da8d

https://github.com/dfinity/js-libp2p-gossip-discovery/commit/4e71f52a2d519ed200cdb9b2788145c5005915b8

https://github.com/dfinity/js-libp2p-gossip-discovery/commit/77060773b66ca38cbbc153cddb8c24dfb7325d46

Closing, and closing related issues.

**Pull request closed without merging**